templates/flutter/lib/client.dart.twig

```diff
@@ -118,18 +108,27 @@ class Client {
             method: method.name(),
         );
 
-        if(headers['content-type'] == 'multipart/form-data') {
-            return http.request(path, data: FormData.fromMap(params), options: options);
-        }
-
-        if (method == HttpMethod.get) {
-            params.keys.forEach((key) {if (params[key] is int || params[key] is double) {
-              params[key] = params[key].toString();
-            }});
-            
-            return http.get(path, queryParameters: params, options: options);
-        } else {
-            return http.request(path, data: params, options: options);
+        try {
+            if(headers['content-type'] == 'multipart/form-data') {
+                return await http.request(path, data: FormData.fromMap(params), options: options);
+            }
+
+            if (method == HttpMethod.get) {
+                params.keys.forEach((key) {if (params[key] is int || params[key] is double) {
+                params[key] = params[key].toString();
+                }});
+                
+                return await http.get(path, queryParameters: params, options: options);
+            } else {
+                return await http.request(path, data: params, options: options);
+            }
+        } on DioError catch(e) {
+          if(e.response == null) {
+            throw {{spec.title | caseUcfirst}}AppwriteException(e.message);
```

Comment:
I think this line in the Exceptions PR is wrong: `throw {{spec.title | caseUcfirst}}AppwriteException(e.message);` as it resolve to `AppwriteAppwriteException`

Reply:
Ah !